sync/category.go

```diff
@@ -0,0 +1,93 @@
+package sync
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/appuio/appuio-cloud-reporting/pkg/db"
+	"github.com/vshn/appuio-odoo-adapter/odoo/model"
+)
+
+// Reconcile synchronizes model.InvoiceCategory in Odoo based on the given db.Category according to the following rules:
+//  * If db.Category.Target is NULL then it will search for an existing category that matches the name:
+//    * If not found, it will create a new model.InvoiceCategory and set db.Category.Target to the ID returned by Odoo.
+//    * If found a match, it will use the model.InvoiceCategory and set db.Category.Target to the ID returned by Odoo.
+//  * If db.Category.Target has a value then it will search for a matching model.InvoiceCategory:
+//    * If not found, it will recreate the model.InvoiceCategory.
+//    * If found and model.InvoiceCategory.Name is up-to-date, it will return without error
+//    * If found and model.InvoiceCategory.Name differs from db.Category.Source, the model.InvoiceCategory is updated/reset.
+func (s *InvoiceCategoryReconciler) Reconcile(ctx context.Context, category *db.Category) error {
```

Comment:
I would personally prefer if `Reconcile` returned the modified category instead of modifying arguments.